frontend/www/js/omegaup/components/identity/Edit.vue

```diff
@@ -0,0 +1,164 @@
+                 for="username">{{ T.username }}</label>
+            <div class="col-md-7 col-sm-7">
+              <div class="input-group">
+                <span class="input-group-addon">{{ groupName }}:</span> <input class="form-control"
+                     name="username"
+                     size="30"
+                     type="text"
+                     v-model="identityName">
+              </div>
+            </div>
+          </div>
+          <div class="form-group">
+            <label class="col-md-4 col-sm-4 control-label"
+                 for="name">{{ T.profile }}</label>
+            <div class="col-md-7 col-sm-7">
+              <input class="form-control"
+                   name="name"
+                   size="30"
+                   type="text"
+                   v-model="identity.name">
+            </div>
+          </div>
+          <div class="form-group">
+            <label class="col-md-4 col-sm-4 control-label"
+                 for="countryId">{{ T.userEditCountry }}</label>
+            <div class="col-md-7 col-sm-7">
+              <select class="form-control"
+                   name="countryId"
+                   v-model="selectedCountry">
+                <option v-bind:value="country.country_id"
+                        v-for="country in countries">
+                  {{ country.name }}
+                </option>
+              </select>
+            </div>
+          </div>
+          <div class="form-group">
+            <label class="col-md-4 col-sm-4 control-label"
+                 for="stateId">{{ T.profileState }}</label>
+            <div class="col-md-7 col-sm-7">
+              <select class="form-control"
+                   name="stateId"
+                   v-model="selectedState">
+                <option v-bind:value="state.code.split('-')[1]"
+                        v-for="state in countryStates"
+                        v-if="countryStates.length &gt; 0">
+                  {{ state.name }}
+                </option>
+              </select>
+            </div>
+          </div>
+          <div class="form-group">
+            <label class="col-md-4 col-sm-4 control-label"
+                 for="school">{{ T.profileSchool }}</label>
+            <div class="col-md-7 col-sm-7">
+              <input class="form-control"
+                   name="school"
+                   size="20"
+                   type="text"
+                   v-model="identity.school">
+            </div><input name="schoolId"
+                 type="hidden"
+                 v-bind:value="identity.school_id">
+          </div>
+        </div>
+        <div class="form-group pull-right">
+          <button class="btn btn-primary"
+               type="submit">{{ T.wordsSaveChanges }}</button> <button class="btn btn-secundary"
+               type="reset"
+               v-on:click="onCancel">{{ T.wordsCancel }}</button>
+        </div>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script>
```

Comment:
puede ser

```suggestion
<script lang="ts">
```

? :D

Reply:
No me funcionó bien. Tengo algunos errores. El principal que no pude corregir fue este:
```
TS7016: Could not find a declaration file for module '../../../../third_party/js/iso-3166-2.js/iso3166.min.js'. '/opt/omegaup/frontend/www/third_party/js/iso-3166-2.js/iso3166.min.js' implicitly has an 'any' type.
``` 
Este me aparece cuando intento importar `iso3166`. No he encontrado la forma de arreglarlo.

Hay otros errores que seguiré revisando

Reply:
sí, eso es esperado. eso del "declaration file" viene explicado acá: https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html

el API de ese módulo está aquí: https://github.com/omegaup/iso-3166-2.js/blob/master/functions.js

aquí hay un ejemplo de cómo declaré el `.d.ts`para [omegaup.js](https://github.com/omegaup/omegaup/blob/4e525c99987de8a4dec7f9681be73b0fe261fe5c/frontend/www/js/omegaup/omegaup.js): https://github.com/omegaup/omegaup/blob/4e525c99987de8a4dec7f9681be73b0fe261fe5c/frontend/www/js/omegaup/omegaup.d.ts

entonces lo que hay que hacer es crear un archivo `frontend/www/js/omegaup/iso-3166-2.d.ts` que contenga

```typescript
declare module '../../../../third_party/js/iso-3166-2.js/iso3166.min.js' {
  interface Subdivision {
    type: string;
    name: string;
  }
  interface Subdivisions {
    [code: string]: Subdivision;
  }

  interface Country {
    name: string;
    sub: Subdivisions;
  }

  export function country(name: string): Country;
}
```

y agregarle más métodos hasta que se deje de quejar.

Reply:
Lo agregué, pero me sigue mandando el mismo error. No sé que es lo que estoy agregando incorrectamente

Reply:
hmmm... a ver, intenta primero cambiar la ruta en el archivo a ser `'../../third_party/js/iso-3166-2.js/iso3166.min.js'`, para que concuerde con el archivo.

si eso no se deja, mueve el archivo a `frontend/www/js/omegaup/components/identity/iso-3166-2.d.ts` para que concuerde con cómo lo importas.

Reply:
No me funciona de ninguna de esas formas. Intenté moverlo a estas tres rutas:
- `frontend/www/js/omegaup/components/identity/iso-3166-2.d.ts`
- `frontend/www/js/omegaup/iso-3166-2.d.ts`
- `frontend/www/third_party/js/iso-3166-2.js/iso-3166-2.d.ts`

Además intenté con este otro nombre:
- `frontend/www/js/omegaup/components/identity/iso3166.d.ts`
- `frontend/www/js/omegaup/iso3166.d.ts`
- `frontend/www/third_party/js/iso-3166-2.js/iso3166.d.ts`.

Sigo sin entender bien como es que se realiza el mapeo.

Otra prueba que intenté es quitar el declaration file de `omegaup.js`, y me encontré con que me aparece otro error distinto:
```
ERROR in /opt/omegaup/frontend/www/js/omegaup/components/RankTable.vue.ts
    [tsl] ERROR in /opt/omegaup/frontend/www/js/omegaup/components/RankTable.vue.ts(89,19)
          TS2306: File '/opt/omegaup/frontend/www/js/omegaup/omegaup.d.ts' is not a module.
    
    ERROR in /opt/omegaup/frontend/www/js/omegaup/components/RankTable.vue.ts
    [tsl] ERROR in /opt/omegaup/frontend/www/js/omegaup/components/RankTable.vue.ts(91,25)
          TS2306: File '/opt/omegaup/frontend/www/js/omegaup/omegaup.d.ts' is not a module.
    
    ERROR in /opt/omegaup/frontend/www/js/omegaup/components/identity/Edit.vue.ts
    [tsl] ERROR in /opt/omegaup/frontend/www/js/omegaup/components/identity/Edit.vue.ts(92,19)
          TS2306: File '/opt/omegaup/frontend/www/js/omegaup/omegaup.d.ts' is not a module.
```

, yo esperaba ver el mismo:
```
ERROR in /opt/omegaup/frontend/www/js/omegaup/components/identity/Edit.vue.ts
    [tsl] ERROR in /opt/omegaup/frontend/www/js/omegaup/components/identity/Edit.vue.ts(93,21)
          TS7016: Could not find a declaration file for module '../../../../third_party/js/iso-3166-2.js/iso3166.min.js'. '/opt/omegaup/frontend/www/third_party/js/iso-3166-2.js/iso3166.min.js' implicitly has an 'any' type.
```

Así que no sé si el problema es algo que no esté considerando.

Reply:
cuando lo moviste a `frontend/www/third_party/js/iso-3166-2.js/iso-3166-2.d.ts`, qué nombre de módulo usaste? `'./iso-3166-2.js'`?

otra pregunta, estás reiniciando `yarn run dev` cada vez que agregas/mueves/eliminas un archivo? resulta que ese comando no puede reconocer bien cuando el conjunto de archivos cambia (sólo reconoce bien si el conjunto es el mismo, pero el contenido de los archivos cambia).

Reply:
Estaba usando este nombre: `iso3166.min.js`, ahora lo intenté con `iso-3166-2.js` además de mover el archivo a otra ubicación y encontré lo siguiente:

- El `import` del archivo `Edit.vue` debe quedar de la siguiente forma: `'../../../../third_party/js/iso-3166-2.js'`y la ubicación donde detectó el archivo es en: `frontend/www/third_party/js/iso-3166-2.d.ts`. con estos cambios ahora me manda el siguiente error:
```
ERROR in /opt/omegaup/frontend/www/js/omegaup/components/identity/Edit.vue.ts
    [tsl] ERROR in /opt/omegaup/frontend/www/js/omegaup/components/identity/Edit.vue.ts(93,21)
          TS2306: File '/opt/omegaup/frontend/www/third_party/js/iso-3166-2.d.ts' is not a module.
```

Ahora comenzaré a investigar como resolver esta falla.

Reply:
aaah aguanta, perdón, todo lo que dije debería tener `.min.js` o `.min.d.ts`. sorry por la confusión u_u

Reply:
ok, deja intento agregarle el `min`

Reply:
Listo, ya le agregué el `.min.js` y `min.d.ts` en sus respectivos lugares. Sigo obteniendo el mismo error:
```
ERROR in /opt/omegaup/frontend/www/js/omegaup/components/identity/Edit.vue.ts
    [tsl] ERROR in /opt/omegaup/frontend/www/js/omegaup/components/identity/Edit.vue.ts(93,21)
          TS2306: File '/opt/omegaup/frontend/www/third_party/js/iso-3166-2.js/iso3166.min.d.ts' is not a module.
    
    ERROR in /opt/omegaup/frontend/www/third_party/js/iso-3166-2.js/iso3166.min.d.ts
    [tsl] ERROR in /opt/omegaup/frontend/www/third_party/js/iso-3166-2.js/iso3166.min.d.ts(14,3)
          TS1128: Declaration or statement expected.
    
    ERROR in /opt/omegaup/frontend/www/third_party/js/iso-3166-2.js/iso3166.min.d.ts
    [tsl] ERROR in /opt/omegaup/frontend/www/third_party/js/iso-3166-2.js/iso3166.min.d.ts(14,17)
          TS1005: ';' expected.
    
    ERROR in /opt/omegaup/frontend/www/third_party/js/iso-3166-2.js/iso3166.min.d.ts
    [tsl] ERROR in /opt/omegaup/frontend/www/third_party/js/iso-3166-2.js/iso3166.min.d.ts(14,29)
          TS1005: ',' expected.
    
    ERROR in /opt/omegaup/frontend/www/third_party/js/iso-3166-2.js/iso3166.min.d.ts
    [tsl] ERROR in /opt/omegaup/frontend/www/third_party/js/iso-3166-2.js/iso3166.min.d.ts(14,38)
          TS1005: ';' expected.
``` 

Lo que me indican la mayoría de los errores es que ya está leyendo el archivo, pero cuando hago modificaciones no se ven reflejados, aún si vuelvo a ejecutar `yarn run dev`. El archivo lo estoy modificando directamente en la VM y sigue sin cambiar nada. Lo que si estoy seguro es que está leyendo el archivo que modifico, porque en cuanto lo borro o le cambio el nombre, en `yarn` me aparece el error de que no encuentra el declaration file:
```
ERROR in /opt/omegaup/frontend/www/js/omegaup/components/identity/Edit.vue.ts
    [tsl] ERROR in /opt/omegaup/frontend/www/js/omegaup/components/identity/Edit.vue.ts(93,26)
          TS7016: Could not find a declaration file for module '../../../../third_party/js/iso-3166-2.js/iso3166.min.js'. '/opt/omegaup/frontend/www/third_party/js/iso-3166-2.js/iso3166.min.js' implicitly has an 'any' type.
```

Reply:
de pura casualidad no tendrás archivos `.d.ts` de más por ahí volando? me suena a que eso es lo que estaría causando que no le haga caso a tus modificaciones.

también, si le agregas un par de líneas en blanco al archivo al principio, los números de línea de los errores cambian?

Reply:
Ya descarté esos dos casos, en `/opt/omegaup` tengo esto:
```
vagrant@ubuntu-bionic:/opt/omegaup$ git status
On branch manage-identities-group-ui
Your branch is up to date with 'origin/manage-identities-group-ui'.

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)
  (commit or discard the untracked or modified content in submodules)

	modified:   frontend/www/js/omegaup/components/RankTable.vue
	modified:   frontend/www/js/omegaup/components/identity/Edit.vue
	deleted:    frontend/www/js/omegaup/iso-3166-2.d.ts
	modified:   frontend/www/third_party/js/iso-3166-2.js (untracked content)
```

y en `/opt/omegaup/frontend/www/third_party/js/iso-3166-2.js` tengo sólo un archivo:
```
vagrant@ubuntu-bionic:/opt/omegaup/frontend/www/third_party/js/iso-3166-2.js$ git status
HEAD detached at 96d7d90
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	iso3166.min.d.ts

nothing added to commit but untracked files present (use "git add" to track)
```

Lo de las líneas en blanco también ya lo intenté y ahí fue donde me di cuenta que no está haciendo caso a mis cambios.